### ISSUE TITLE
🍎 Neighbourhood and districts tooltips

### ISSUE THREAD

**Issue by vogelino** [vogelino/data-2-resilience#21]
The layer selection UI should either allow Neighbourhood or districts to be displayed. When a district or neighbourhood is hovered, a tooltip should show its name and whether it is a risk area or not. Risk-area should be highlighted.

![Image](https://github.com/user-attachments/assets/2c39cd6b-5ef2-4dd3-82c7-bd68bf4a000c)

**Comment by alsino** [comment]
Yes! For that, I would replace the current logic of loading the two geojson layers separately (district/LORs) and just load one unified topojson layer (file attached).

This way we have a smaller file size ad shared boundaries. The topojson JS library can then return Geojson from that file like so:

`import * as topojson from "topojson-client";

// Fetch the TopoJSON file
fetch('path/to/your/topojson/file.json')
  .then(response => response.json())
  .then(topoData => {
    // Access "bezirke" and "stadtteile" from the TopoJSON file
    const bezirkeGeoJSON = topojson.feature(topoData, topoData.objects.bezirke);
    const stadtteileGeoJSON = topojson.feature(topoData, topoData.objects.stadtteile);

    // Log or use the GeoJSON objects
    console.log('Bezirke GeoJSON:', bezirkeGeoJSON);
    console.log('Stadtteile GeoJSON:', stadtteileGeoJSON);

    // Example: Use the data in a map
    // map.addSource('bezirke', { type: 'geojson', data: bezirkeGeoJSON });
    // map.addSource('stadtteile', { type: 'geojson', data: stadtteileGeoJSON });
  })
  .catch(error => {
    console.error('Error loading TopoJSON file:', error);
  });
`


[dortmund-topo.json](https://github.com/user-attachments/files/18318310/dortmund-topo.json)

**Comment by alsino** [comment]
Hab die neuen Daten jetzt schonmal integriert. Fehlen also nur noch die Tooltips ;)

**Comment by alsino** [comment]
Very nice! One little detail: I have renamed the label "Vulnerableraum" to "Vulnerabler Aktionsraum" (Vulnerable action area). 

One thing: In the local version I get the tooltip with extra label "Sozialraum", which I think is nice, but on the deployed version this is missing. Can we add this again maybe?

<img width="235" alt="Image" src="https://github.com/user-attachments/assets/a57e42e7-330c-4cab-b95e-350f7d7c39e3" />
<img width="235" alt="Image" src="https://github.com/user-attachments/assets/c107fbea-9bf5-40db-91d8-aea70a5acdb4" />

**Comment by vogelino** [comment]
> One thing: In the local version I get the tooltip with extra label "Sozialraum", which I think is nice, but on the deployed version this is missing. Can we add this again maybe?

Good point. Weird. I'll have a look!

**Comment by alsino** [comment]
So, we have learned see that there are two different hierarchical levels that interest us:
a) Stadtbezirke (districts)
b) Stadtteile (neighborhoods)

The "Stadtteile" include the vulnerable "Aktionsräume" (action areas). However, they are not always congruent, so in a few cases an action area overlaps two neighborhoods or draws from an even smaller unit called "Unterbezirk" (subdistrict).

I would suggest to show district boundaries as before and when we select "Aktionsräume" with the radio button we show the shapes of both "Stadtteile" and "Aktionsräume", but only make "Aktionsräume interactive on hover (see screenshot).

What do you think?


<img width="1277" alt="Image" src="https://github.com/user-attachments/assets/2dc4cb23-493a-4584-97bc-af0745a382bd" />

**Comment by alsino** [comment]
I have already added the new geo data (https://github.com/vogelino/data-2-resilience/commit/08e51d4a0afd596bab48e6845d6d8e479476a782) including the three relevant layers. They are available for display in: **mapData.ts**

```javascript
export const districts = topojson.feature(
	// @ts-ignore
	dortmundGeodata,
	dortmundGeodata.objects.bezirke
);
export const lors = topojson.feature(
	// @ts-ignore
	dortmundGeodata,
	dortmundGeodata.objects.stadtteile
);

export const actionAreas = topojson.feature(
	// @ts-ignore
	dortmundGeodata,
	dortmundGeodata.objects.aktionsraeume
);
```

**Comment by alsino** [comment]
The new combined data is here: 
[dortmund-topo.json](https://github.com/user-attachments/files/18451312/dortmund-topo.json)

It can be displayed like so:

<img width="508" alt="Image" src="https://github.com/user-attachments/assets/76e17db9-6c98-421d-8c1e-c23f2a0aa900" />

**Comment by vogelino** [comment]
Hey Alsino!

Thx for taking the time to prepare the topojson and to thoroughly explaining the difference. I think yours is a good solution and will probably convince de RUB Team. 

I'll integrate your solition an will let you know when it's live. 

Cheers,

Lucas

**Comment by vogelino** [comment]
@alsino Tooltips for both layer types are added and the topojson was updated. Check it out and let me know what you think. Cheers!

PS: Related to / Includes #35

**Comment by alsino** [comment]
I think it looks great and works as expected! 👍

One little thing: Maybe for the "Aktionräume" layer, in the tooltip we could replace the "Aktionsraum" (en: "Action area") with "Stadtteil" (en: "Neighborhood") for the ones that are not yellow, as Aktionnsrraum really only refers to the yellow areas.

<img width="299" alt="Image" src="https://github.com/user-attachments/assets/9c7416f5-6a26-4fee-b913-915a9cf0cfd8" />

**Comment by vogelino** [comment]
> we could replace the "Aktionsraum" (en: "Action area") with "Stadtteil" (en: "Neighborhood") for the ones that are not yellow

Good point! Will do!

**Comment by vogelino** [comment]
> we could replace the "Aktionsraum" (en: "Action area") with "Stadtteil" (en: "Neighborhood") for the ones that are not yellow

Good point! Will do!
